prometheus/counter.go

```diff
@@ -121,6 +123,10 @@ func (c *counter) Add(v float64) {
 		panic(errors.New("counter cannot decrease in value"))
 	}
 
+	if c.createdTs == nil {
```

Comment:
I think this is incorrect. The metric is created on `WithLabelValues` and family. This is leveraged when people are dealing with init problem now. They do:

1. Create metric counterVec (e.g. with label "code")
2. Register it
3. do `metric.WithLabelValues("200")` (no inc or add)

This behaviour is confirmed by [docs](https://github.com/prometheus/client_golang/blob/main/prometheus/counter.go#L215): 

```
// GetMetricWithLabelValues returns the Counter for the given slice of label
// values (same order as the variable labels in Desc). If that combination of
// label values is accessed for the first time, a new Counter is created.
```

(WithLabelValues and family invokes GetMetricWithLabelValues)

It feels we should add timestamp on GetMetricWithLabelValues of MetricVec or deeper/shallower, not when the metric gets incremented? 

Also test for that would be epic (another case)

Reply:
Oh I didn't know that, thanks for the info :) 
I'll fix and add a test case.

But what about counters that are not part of a CounterVec? The only way we know they are initialized is once they are incremented for the first time, isn't it? Should we keep that conditional in `Add()` and `Inc()` and just add another to `WithLabelValues`?

Reply:
Ok, just saw https://github.com/prometheus/client_golang/pull/1044. Looks like the correct implementation :P